### PR TITLE
Tweak error messages

### PR DESCRIPTION
**Pull request by krlmlr**
from https://github.com/tidyverse/tibble/pull/731.

### CODE REVIEW

File: tests/testthat/test-slice.R [hunk 1]
```diff
@@ -191,7 +191,7 @@ test_that("vec_as_location() checks type", {
   expect_error(vec_as_location("foo", "bar"), class = "vctrs_error_incompatible_type")
   expect_error(vec_as_location("foo", 1L, names = 1L), "must be a character vector")
   expect_error(vec_as_location(Sys.Date(), 3L), class = "vctrs_error_subscript_type")
-  expect_error(vec_as_location(matrix(TRUE, nrow = 1), 3L), "must have one dimension")
+  expect_error(vec_as_location(matrix(TRUE, nrow = 1), 3L), "must be a one-dimensional")
```
Comment (lionel-) [inline tests/testthat/test-slice.R]: Can you add this to the error output file please? And wrap this test in `verify_errors()`.

File: tests/testthat/test-slice.R [hunk 2]
```diff
@@ -431,6 +430,10 @@ test_that("slice has informative error messages", {
     "# Can't index beyond the end of a vector"
     vec_slice(1:2, 3L)
     vec_slice(1:2, -3L)
+
+    "# Can't be matrix"
```
Comment (lionel-) [inline tests/testthat/test-slice.R]: I don't see the corresponding `verify_errors()`. The title here should be the same title as the `test_that()` block where the verify-errors is located.

**Comment by lionel-** [comment]
```
  -- 1. Failure: (unknown) (@test-conditions.R#101)  -----------------------------
  Results have changed from known value recorded in 'error/test-conditions.txt'.
  Lengths differ: 87 is not 88
```

File: tests/testthat/test-slice.R [hunk 3]
```diff
@@ -431,8 +437,10 @@ test_that("slice has informative error messages", {
     vec_slice(1:2, 3L)
     vec_slice(1:2, -3L)
 
-    "# Can't be matrix"
+    "# vec_slice throws error with non-vector subscripts"
     vec_slice(1:3, matrix(TRUE, ncol = 1))
```
Comment (lionel-) [inline tests/testthat/test-slice.R]: You're either missing checks here, or you should pull them out of the `verify_errors()` block. The goal is to keep verify-errors blocks in sync with the corresponding block in the output check.

**Comment by krlmlr** [comment]
I have no idea where the testing error comes from. I have most recent rlang.

**Comment by lionel-** [comment]
This error should now be fixed on master. Thanks!

**Pull request merged**